src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,172 @@
+import classNames from 'classnames';
+import { useCallback, useMemo } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink';
+import { SearchLink } from '../SearchLink';
+import { headers } from '../../utils/headers';
+
+type Props = {
+  people: Person[]
+  selectedPerson: string
+};
+
+export const PeopleTable: React.FC<Props> = ({ people, selectedPerson }) => {
+  const activePerson = people.find(
+    person => person.slug === selectedPerson,
+  );
+
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+  const personsGender = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const visiblePeople = useMemo(() => {
+    return [...people].sort((person1, person2) => {
+      switch (sort) {
+        case 'name':
+          return order
+            ? person2.name.localeCompare(person1.name)
+            : person1.name.localeCompare(person2.name);
+        case 'sex':
+          return order
+            ? person2.sex.localeCompare(person1.sex)
+            : person1.sex.localeCompare(person2.sex);
+        case 'born':
+          return order
+            ? person2.born - person1.born
+            : person1.born - person2.born;
+        case 'died':
+          return order
+            ? person2.died - person1.died
+            : person1.died - person2.died;
+        default:
+          return 0;
+      }
+    }).filter(({
+      sex, name, fatherName, motherName, born,
+    }) => (!personsGender || sex === personsGender)
+      && (!centuries.length
+        || centuries.includes((born + 100).toString().slice(0, 2)))
+      && (name.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+      || fatherName?.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+      || motherName?.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+      ));
```

Comment:
a little bit complicated 
create separate function for this logic and use it in useMemo

Reply:
also try to move all this long conditions into variables